packages/cdk8s-cli/lib/import/type-generator.ts

```diff
@@ -92,7 +92,10 @@ export class TypeGenerator {
   }
 
   public emitType(typeName: string, def: JSONSchema4, structFqn: string): string {
-    typeName = normalizeTypeName(typeName);
+
+    if (normalizeTypeName(typeName) !== typeName) {
```

Comment:
Because then the *actual* type name will then not be the same as the value of `typeName` passed from the call site. This means that the caller can't assume that the type emitted actually has the expected name, and it cannot use this name.

For example, the code that invokes `emitType` to emit the `XxxOptions` struct looks like this (sketch):

```ts
const typeName = constructName + 'Options';
emitType(typeName, optionsStructSchema);

emit(`constructor(scope: Construct, id: string, options: ${typeName}) { ... }`);
```

This means that if `typeName` is mangled within `emitType`, the external value is wrong.

Reply:
I'll add a comment to the code.

Reply:
Fixed, and realized the class name itself was not normalized as well. Thanks!